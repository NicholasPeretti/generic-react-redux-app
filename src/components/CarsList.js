import React from 'react'
import PropTypes from 'prop-types'
import CarsListItem from './CarsListItem'
import styled from 'styled-components'
import CarPropType from '../utils/propTypes/car'
import SubTitle from './Title'

const ListContainer = styled.div`
  display: flex;
  flex-direction: column;
  width: 100%;
`
const Center = styled.div`
  display: flex;
  width: 100%;
  justify-content: center;
`

var CarsList = ({ cars }) => {
  if (!cars.length) {
    return (
      <Center>
        <SubTitle>No cars found</SubTitle>
      </Center>
    )
  }
  return (
    <ListContainer>
      {cars.map((car, key) => (
        <CarsListItem car={car} key={key} />
      ))}
    </ListContainer>
  )
}

CarsList.propTypes = {
  cars: PropTypes.arrayOf(CarPropType).isRequired
}

export default CarsList
