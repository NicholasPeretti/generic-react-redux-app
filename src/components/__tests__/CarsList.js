import React from 'react'
import { shallow } from 'enzyme'
import CarsList from '../CarsList'
import PresentationalTest from '../../utils/tests/presentational'
import { MemoryRouter } from 'react-router-dom'

const WrappedCarsList = props => (
  <MemoryRouter>
    <CarsList {...props} />
  </MemoryRouter>
)

const props = {
  cars: [{
    stockNumber: 86987,
    manufacturerName: 'Fiat',
    modelName: 'Talento',
    color: 'white',
    mileage: {
      number: 102586,
      unit: 'km'
    },
    fuelType: 'Petrol',
    pictureUrl: 'http://localhost:3001/car.svg'
  }]
}

describe('CarsList', () => {
  PresentationalTest(WrappedCarsList, props)

  const wrapper = shallow(<WrappedCarsList {...props} />)

  it('Should match snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })
})
