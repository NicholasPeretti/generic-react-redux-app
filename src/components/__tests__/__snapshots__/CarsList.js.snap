// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CarsList Should match snapshot 1`] = `"<div class=\\"sc-htoDjs cfVSXB\\"><article class=\\"sc-bZQynM hEXood\\"><img src=\\"http://localhost:3001/car.svg\\" class=\\"sc-EHOje qtPkP\\"/><section class=\\"sc-ifAKCX hPHZRJ\\"><h3 class=\\"sc-htpNat sc-bxivhb cjMArm\\">Fiat Talento</h3><span class=\\"sc-bdVaJa bkmfbd\\">Stock # 86987 - 102586 km - Petrol - white</span><span class=\\"sc-bdVaJa bkmfbd\\"><a class=\\"sc-bwzfXH isCRHY\\" href=\\"/86987\\">View details</a></span></section></article></div>"`;
